apps/webhook/src/webhooks/usecases/execution-details/execution-details.usecase.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable, Module } from '@nestjs/common';
+import { ExecutionDetailsEntity, ExecutionDetailsRepository, MessageEntity } from '@novu/dal';
+import { ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+
+import { ExecutionDetailsCommand, WebhookCommand } from './execution-details.command';
+
+import { IWebhookResult } from '../../dtos/webhooks-response.dto';
+
+@Injectable()
+export class ExecutionDetails {
+  constructor(private executionDetailsRepository: ExecutionDetailsRepository) {}
+
+  async execute(command: ExecutionDetailsCommand): Promise<void> {
+    const executionDetailsEntity = this.mapWebhookEventIntoEntity(
+      command.webhook,
+      command.message,
+      command.webhookEvent
+    );
+
+    // TODO: Do we want to let this bubble up and break the webhook use case flow?
+    await this.executionDetailsRepository.create(executionDetailsEntity);
```

Comment:
Do you mean in case of a failure here? I think we can bubble it up, because the flow is not really complete if this is not saved.

Reply:
But in the case of multiple events for the webhook, the parsing would be aborted in the middle of the flow if one fails when creating the execution details. That would also mean having partial data as we wouldn't delete the execution details of the previous successful details (full rollback).
Not sure what's the strategy we want to have here for this flow thus raising the question.